docs/exts/docs_build/run_patched_sphinx.py

```diff
@@ -0,0 +1,220 @@
+        file_patterns = app.config.autoapi_file_patterns
+    else:
+        file_patterns = DEFAULT_FILE_PATTERNS.get(app.config.autoapi_type, [])
+
+    if app.config.autoapi_ignore:
+        ignore_patterns = app.config.autoapi_ignore
+    else:
+        ignore_patterns = DEFAULT_IGNORE_PATTERNS.get(app.config.autoapi_type, [])
+
+    if ".rst" in app.config.source_suffix:
+        out_suffix = ".rst"
+    elif ".txt" in app.config.source_suffix:
+        out_suffix = ".txt"
+    else:
+        # Fallback to first suffix listed
+        out_suffix = app.config.source_suffix[0]
+
+    if sphinx_mapper_obj.load(patterns=file_patterns, dirs=normalised_dirs, ignore=ignore_patterns):
+        sphinx_mapper_obj.map(options=app.config.autoapi_options)
+
+        if app.config.autoapi_generate_api_docs:
+            sphinx_mapper_obj.output_rst(root=normalized_root, source_suffix=out_suffix)
+
+        if app.config.autoapi_type == "python":
+            app.env.autoapi_objects = sphinx_mapper_obj.objects
+            app.env.autoapi_all_objects = sphinx_mapper_obj.all_objects
+
+
+# HACK: sphinx-auto map did not correctly use the confdir attribute instead of srcdir when specifying the
+# directory to contain the generated files.
+# Unfortunately we have a problem updating to a newer version of this library and we have to use
+# sphinx-autoapi v1.0.0, so I am monkeypatching this library to fix this one problem.
+autoapi.extension.run_autoapi = run_autoapi
+
+
+def _resolve_module_placeholders(modules, module_name, visit_path, resolved):
+    """Resolve all placeholder children under a module.
+
+    :param modules: A mapping of module names to their data dictionary.
+        Placeholders are resolved in place.
+    :type modules: dict(str, dict)
+    :param module_name: The name of the module to resolve.
+    :type module_name: str
+    :param visit_path: An ordered set of visited module names.
+    :type visited: collections.OrderedDict
+    :param resolved: A set of already resolved module names.
+    :type resolved: set(str)
+    """
+    if module_name in resolved:
+        return
+
+    visit_path[module_name] = True
+
+    module, children = modules[module_name]
+    for child in list(children.values()):
+        if child["type"] != "placeholder":
+            continue
+
+        if child["original_path"] in modules:
+            module["children"].remove(child)
+            children.pop(child["name"])
+            continue
+
+        imported_from, original_name = child["original_path"].rsplit(".", 1)
+        if imported_from in visit_path:
+            msg = f"Cannot resolve cyclic import: {', '.join(visit_path)}, {imported_from}"
+            mapper.LOGGER.warning(msg)
+            module["children"].remove(child)
+            children.pop(child["name"])
+            continue
+
+        if imported_from not in modules:
+            if not imported_from.startswith("airflow.providers") or module_name.startswith(
```

Comment:
Here is the only change to this function. I added this if statement.

Reply:
If we need to keep this function could you "bracket" this added/changed section with a `# begin change`, `# end change` please?